File: 프로그래머스/1/12922. 수박수박수박수박수박수？/수박수박수박수박수박수？.js
function solution(n) {
    // 1번째 풀이
    // let answer = '수'
    // for(let i = 2; i <= n; i++){
    //     i%2 === 0 ? answer += '박' : answer += '수'
    // }
    // return answer
    
    // 2번째 풀이
    // n길이 만큼의 배열을 생성한다.
    // 인덱스가 짝수일 경우 수, 홀수일 경우 박을 배열에 담는다.
    // join()을통해 배열을 문자열로 바꾼다.
    return Array.from({length: n}, (_,id)=>{
        return id%2 == 0 ? '수' : '박'
    }).join('')
}

File: 프로그래머스/1/12922. 수박수박수박수박수박수？/수박수박수박수박수박수？.py
def solution(n):
    # 삼항 연산자와 리스트 컴프리헨션을 이용하여 리스트를 만든다.
    list = ["수" if i % 2 == 0 else "박"  for i in range(n)]
    # 문자열의 join 매서드를 활용하여 리스트를 문자열로 만든다.
    return ''.join(list)

